types/d3-array/d3-array-tests.ts

```diff
@@ -364,6 +371,70 @@ numOrUndefined = d3Array.scan(readonlyMixedObjectOrUndefinedArray, (a, b) => {
     return a && b ? a.num - b.num : NaN;
 });
 
+// leastIndex() ----------------------------------------------------------------------
+
+numOrUndefined = d3Array.least(numbersArray);
+numOrUndefined = d3Array.least(typedArray);
+numOrUndefined = d3Array.least(readonlyNumbersArray);
+
+let mo: MixedObject | undefined = d3Array.least(mixedObjectArray, (a, b) => {
+    const aElem: MixedObject = a;
+    const bElem: MixedObject = b;
+    return a.num - b.num;
+});
+
+mo = d3Array.least(readonlyMixedObjectArray, (a, b) => {
+    const aElem: MixedObject = a;
+    const bElem: MixedObject = b;
+    return a.num - b.num;
+});
+
+mo = d3Array.least(mixedObjectOrUndefinedArray, (a, b) => {
+    const aElem: MixedObject | undefined = a;
+    const bElem: MixedObject | undefined = b;
+    return a && b ? a.num - b.num : NaN;
+});
+
+mo = d3Array.least(readonlyMixedObjectOrUndefinedArray, (a, b) => {
+    const aElem: MixedObject | undefined = a;
+    const bElem: MixedObject | undefined = b;
+    return a && b ? a.num - b.num : NaN;
+});
+
+mo = d3Array.least(readonlyMixedObjectOrUndefinedArray, a => a ? a.num : null);
+
+// leastIndex() ----------------------------------------------------------------------
+
+numOrUndefined = d3Array.leastIndex(numbersArray);
```

Comment:
Had a small hunch that the following could be better since as per docs, the output of `.leastIndex` is a number.

```
num = d3Array.leastIndex(numbersArray);
```